e2e/e2e_canary_upgrade_test.go

```diff
@@ -0,0 +1,290 @@
+)
+
+var _ = Describe("Fleetshard-sync Targeted Upgrade", func() {
+
+	BeforeEach(func() {
+		if !runCanaryUpgradeTests {
+			Skip("Skipping canary upgrade test")
+		}
+	})
+
+	if fmEndpointEnv := os.Getenv("FLEET_MANAGER_ENDPOINT"); fmEndpointEnv != "" {
+		fleetManagerEndpoint = fmEndpointEnv
+	}
+
+	option := fleetmanager.OptionFromEnv()
+	auth, err := fleetmanager.NewStaticAuth(fleetmanager.StaticOption{StaticToken: option.Static.StaticToken})
+	Expect(err).ToNot(HaveOccurred())
+	client, err := fleetmanager.NewClient(fleetManagerEndpoint, auth)
+	Expect(err).ToNot(HaveOccurred())
+
+	Describe("should run ACS operators", func() {
+
+		It("should deploy single operator", func() {
+			ctx := context.Background()
+			oneOperatorVersionConfig := []operator.OperatorConfig{{
+				GitRef: "4.1.1",
+				Image:  "quay.io/rhacs-eng/stackrox-operator:4.1.1",
+			}}
+			err := updateOperatorConfig(ctx, oneOperatorVersionConfig)
+			Expect(err).To(BeNil())
+
+			Eventually(validateOperatorDeployment(ctx, "4.1.1", "quay.io/rhacs-eng/stackrox-operator:4.1.1")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+
+		})
+
+		It("should deploy two operators", func() {
+			ctx := context.Background()
+			twoOperatorVersionConfig := []operator.OperatorConfig{{
+				GitRef: "4.1.1",
+				Image:  "quay.io/rhacs-eng/stackrox-operator:4.1.1",
+			},
+				{
+					GitRef: "4.1.2",
+					Image:  "quay.io/rhacs-eng/stackrox-operator:4.1.2",
+				},
+			}
+
+			err := updateOperatorConfig(ctx, twoOperatorVersionConfig)
+			Expect(err).To(BeNil())
+
+			Eventually(validateOperatorDeployment(ctx, "4.1.1", "quay.io/rhacs-eng/stackrox-operator:4.1.1")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+			Eventually(validateOperatorDeployment(ctx, "4.1.2", "quay.io/rhacs-eng/stackrox-operator:4.1.2")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+			Eventually(getOperatorDeployments(ctx)).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(HaveLen(2))
+
+		})
+
+		It("should delete the removed operator", func() {
+			ctx := context.Background()
+			operatorConfig := []operator.OperatorConfig{{
+				GitRef: "4.1.2",
+				Image:  "quay.io/rhacs-eng/stackrox-operator:4.1.2",
+			}}
+			err := updateOperatorConfig(ctx, operatorConfig)
+			Expect(err).To(BeNil())
+
+			Eventually(validateOperatorDeployment(ctx, "4.1.2", "quay.io/rhacs-eng/stackrox-operator:4.1.2")).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(Succeed())
+			Eventually(getOperatorDeployments(ctx)).
+				WithTimeout(waitTimeout).
+				WithPolling(defaultPolling).
+				Should(HaveLen(1))
+
+		})
+
+	})
+
+	Describe("should upgrade the central", func() {
+
+		BeforeEach(func() {
+			Skip("Skipping canary upgrade test")
+		})
+
+		var createdCentral *public.CentralRequest
+		var centralNamespace string
+
+		It("create central", func() {
```

Comment:
Does this have more expectations? for example, creating Central in a specific version?

```suggestion
		It("creates central", func() {
```